src/main/java/com/cascadebot/cascadebot/commandmeta/Argument.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2019 CascadeBot. All rights reserved.
+ * Licensed under the MIT license.
+ */
+
+package com.cascadebot.cascadebot.commandmeta;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class Argument {
+
+    private String arg;
+    private String description;
+    private Set<Argument> subArgs;
+    private ArgumentType type;
+    private Set<String> aliases;
```

Comment:
Since this will effectively be entirely immutable, every field should be set to final